### PR TITLE
Add submap and gravity-aligned frame terminology.

### PR DESCRIPTION
**Pull request by wohe**

### CODE REVIEW

File: docs/source/terminology.rst [hunk 1]
```diff
@@ -35,16 +35,36 @@ local (map) frame
...
+  The frame in which sensor data is expressed. It is not fixed, i.e. changes
+  over time. It is also different for different trajectories.
+
+gravity-aligned frame
+  Only used in 2D. A frame colocated with the tracking frame but with a
+  different orientation which is approximately aligned with gravity, i.e. the
+  gravitational acceleration vector points approximately in the -z direction. No
+  assumption about yaw (rotation around the z axis between this and the tracking
+  frame) should be made. A different gravity-aligned frame is used for different
+  trajectory nodes.
```
Comment (cschuet) [inline docs/source/terminology.rst]: Maybe explicitly call out that the yaw relationship between tracking and gravity aligned can changed arbitrarily between e.g. consecutive. Essentially in time gravity-aligned frames have no yaw relationship to one another.

Comment (ojura) [reply docs/source/terminology.rst]: > between e.g. consecutive.

Have you accidentally a word?

File: cartographer/mapping/submaps.h [hunk 2]
```diff
@@ -53,18 +53,19 @@ inline uint8 ProbabilityToLogOddsInteger(const float probability) {
...
  public:
-  Submap(const transform::Rigid3d& local_pose) : local_pose_(local_pose) {}
+  Submap(const transform::Rigid3d& local_submap_pose)
+      : local_pose_(local_submap_pose) {}
   virtual ~Submap() {}
 
   virtual void ToProto(proto::Submap* proto) const = 0;
 
-  // Local SLAM pose of this submap.
+  // Pose of this submap in the local frame.
```
Comment (damonkohler) [inline cartographer/mapping/submaps.h]: local map frame?

File: cartographer/mapping_3d/submaps.cc [hunk 3]
```diff
@@ -198,8 +198,8 @@ proto::SubmapsOptions CreateSubmapsOptions(
 }
 
 Submap::Submap(const float high_resolution, const float low_resolution,
-               const transform::Rigid3d& local_pose)
-    : mapping::Submap(local_pose),
+               const transform::Rigid3d& local_submap_pose)
```
Comment (damonkohler) [inline cartographer/mapping_3d/submaps.cc]: Is this different than the local map frame?

File: docs/source/terminology.rst [hunk 4]
```diff
@@ -35,16 +35,37 @@ local (map) frame
   point in time, the transform between this and the global map frame may change,
   but the transform between this and all other frames does not change.
 
+submap frame
+  There is a separate fixed submap frame for each submap.
```
Comment (damonkohler) [inline docs/source/terminology.rst]: Each submap has a separate fixed frame.

File: docs/source/terminology.rst [hunk 5]
```diff
@@ -35,16 +35,37 @@ local (map) frame
...
+
 tracking frame
-  The frame in which sensor data is expressed.
+  The frame in which sensor data is expressed. It is not fixed, i.e. changes
```
Comment (damonkohler) [inline docs/source/terminology.rst]: i.e. it changes ...

File: docs/source/terminology.rst [hunk 6]
```diff
@@ -35,16 +35,37 @@ local (map) frame
...
+  over time. It is also different for different trajectories.
+
+gravity-aligned frame
+  Only used in 2D. A frame colocated with the tracking frame but with a
+  different orientation which is approximately aligned with gravity, i.e. the
```
Comment (damonkohler) [inline docs/source/terminology.rst]: s/which/that

File: docs/source/terminology.rst [hunk 7]
```diff
@@ -35,16 +35,37 @@ local (map) frame
...
 
 
 
 Transforms
 ==========
 
 local_pose
-  Transforms data from the tracking frame to the local frame.
+  Transforms data from the tracking frame (or a submap frame, depending on
+  context) to the local frame.
```
Comment (damonkohler) [inline docs/source/terminology.rst]: local map frame? local submap frame?

File: docs/source/terminology.rst [hunk 8]
```diff
@@ -35,16 +35,37 @@ local (map) frame
...
 
 global_pose
-  Transforms data from the tracking frame to the global frame.
+  Transforms data from the tracking frame (or a submap frame, depending on
+  context)  to the global frame.
```
Comment (damonkohler) [inline docs/source/terminology.rst]: Extra space after the parentheses. Also, global map frame?

File: docs/source/terminology.rst [hunk 9]
```diff
@@ -21,30 +21,51 @@ This documents a few common patterns that exist in the Cartographer codebase.
 Frames
 ======
 
-global (map) frame
+global map frame
```
Comment (damonkohler) [inline docs/source/terminology.rst]: Unless you intend to rename the accessors as well, you should mention here that it is sometimes referred to as just global frame or local frame. Either in parens or after a comma for Ctrl+F friendliness.

Comment (wohe) [reply docs/source/terminology.rst]: I'm not sure what you would like to be changed. There is no mention of "global frame" or "local frame" in the whole codebase anymore.

File: cartographer/mapping_3d/submaps.cc (hunk 3 above)
Comment (wohe) [reply cartographer/mapping_3d/submaps.cc]: This is not a frame. The local submap pose according to terminology:
  Transforms data from a submap frame to the local map frame.

**Pull request merged**